pkg/nsx/services/subnet/store.go

```diff
@@ -51,6 +52,34 @@ func subnetSetIndexFunc(obj interface{}) ([]string, error) {
 // SubnetStore is a store for subnet.
 type SubnetStore struct {
 	common.ResourceStore
+	pathLocks map[string]*sync.Mutex
```

Comment:
The check-then-act pattern in the if statement is not atomic. Multiple goroutines could simultaneously determine that the lock doesn't exist and try to create a new one.

Reply:
Consider using sync.Map?

Reply:
Updated to use sync.Map, please take another look, thanks!